app/src/main/java/org/openobservatory/ooniprobe/domain/callback/GetMeasurementsCallback.java

```diff
@@ -3,28 +3,35 @@
 import org.openobservatory.ooniprobe.R;
 import org.openobservatory.ooniprobe.model.api.ApiMeasurement;
 
+import java.io.IOException;
+
+import okhttp3.ResponseBody;
 import retrofit2.Call;
 import retrofit2.Callback;
 import retrofit2.Response;
 
-public abstract class GetMeasurementsCallback implements Callback<ApiMeasurement> {
+public abstract class GetMeasurementsCallback implements Callback<ResponseBody> {
     @Override
-    public void onResponse(Call<ApiMeasurement> call, Response<ApiMeasurement> response) {
+    public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
         if (response.isSuccessful() &&
                 response.body() != null &&
-                response.body().hasResultWithMeasurements()) {
-            onSuccess(response.body().firstResult());
+                response.isSuccessful()) {
```

Comment:
This seems wrong: you're testing twice whether `response.isSuccessful()`. We should drop the second check.

Reply:
sounds good. let me do that.